Tools/build/compute-changes.py

```diff
@@ -0,0 +1,183 @@
+"""Determine which GitHub Actions workflows to run.
+
+Called by ``.github/workflows/reusable-change-detection.yml``.
+We only want to run tests on PRs when related files are changed,
+or when someone triggers a manual workflow run.
+This improves developer experience by not doing (slow)
+unnecessary work in GHA, and saves CI resources.
+"""
+
+from __future__ import annotations
+
+import os
+import subprocess
+from dataclasses import dataclass
+from pathlib import Path
+
+TYPE_CHECKING = False
+if TYPE_CHECKING:
+    from collections.abc import Set
+
+GITHUB_DEFAULT_BRANCH = os.environ.get("GITHUB_DEFAULT_BRANCH", "")
```

Comment:
This doesn't need a fallback since it's always set on invocation. Let it fail loudly if that stops being the case:
```suggestion
GITHUB_DEFAULT_BRANCH = os.environ["GITHUB_DEFAULT_BRANCH]
```